custom_components/samsungtv_tizen/media_player.py

```diff
@@ -284,6 +284,10 @@ def _get_running_app(self):
                         if 'visible' in root:
                             if root['visible']:
                                 return app
+                                
+                for attr, value in self._app_list.items():
```

Comment:
This `for` loop is in another `for` loop, can we move this above the other `for` loop in this method? Because if it works, it won't need to do the requests to get the running app anymore, so it's a more efficient way of checking.

Maybe also add `if hasattr(self, '_cloud_state') and self._cloud_channel_name:` before it so we only try to check this if SmartThings is enabled?

Reply:
That loop being inside the other one was actually just an indentation error on my part. I moved it before the other one now, with an additional check as you suggested